src/main.c

```diff
@@ -41,7 +41,7 @@ void listDir(mtar_t* tar, char* directory, char* subtract) {
       if ((strcmp(de->d_name, ".") != 0) && (strcmp(de->d_name, "..") != 0)) {
         char path[PATH_MAX];
         snprintf(path, PATH_MAX, "%s/%s", directory, de->d_name);
-        char* finalFileName = path + strlen(subtract);
+        char* finalFileName = path + strlen(subtract) + 1;
```

Comment:
What's the purpose of this `+ 1`? Removing it resolves my trimming issue, but it may have additional effects.

Reply:
I think it exists cause we use `subtract`. I'm going to share some changes where I cleaned up that part. So far everything is working, except for one thing: when passing folders, it won't account for the extra trailing slash. I don't know if there's a C standard library or POSIX way to handle this properly, but if there is, I'd rather use that than manually searching for trailing slash).

Reply:
Unfortunately there isn't a standard or POSIX way to do that as far as I know. I think we have to assume that the end of the path string contains any number of trailing slashes and shorten the path accordingly.